### ISSUE TITLE
AzCopy missing ETA volume progress and currently transferred files

### ISSUE THREAD

**Issue by VelizarVESSELINOV** [Azure/azure-storage-azcopy#981]
### Which version of the AzCopy was used? 
AzCopy 10.4.0

### Which platform are you using? (ex: Windows, Mac, Linux)
Mac

### What command did you run?
azcopy sync "https://..." local

### What problem was encountered?
`azcopy rsync` output:
`15000 Files Scanned at Source, 3581 Files Scanned at Destination, 2-sec Throughput (Mb/s): 553.4614`

Some progress but % is unclear files or volume? Don't see the progress [  x.y GiB/  z.s GiB]
`1.4 %, 542 Done, 0 Failed, 15935 Pending, 16477 Total, 2-sec Throughput (Mb/s): 7.1744`

Expected output like gsutil with ETA, volume progress, and which files are currently transferred.
Plus options to compression and work in parallel
`gsutil -m rsync` output
`Copying xyz...
[118/118 files][  2.3 GiB/  2.3 GiB] 100% Done   1.7 MiB/s ETA 00:00:00`

### Missing options:
- [x] m: parallel execution
- [ ] j: compression gzip

**Comment by JohnRusk** [comment]
It's already parallel.  ;-)  You can't really turn that off.

We have the provision of a job-completion ETA on our backlog, but I'm not sure which release it might be in.

I don't think we'll ever make it output which files are currently transferred, sorry. But there might be a workaround. Let me check with a colleague: @zezha-msft is it safe and supported to use Azcopy Jobs Show on a job that is currently running? 

As for compression, what would you expect there @VelizarVESSELINOV? Would you expect the resulting blob, in Azure storage, to be compressed?  Or not?  E.g. if you look at the blob in some other tool - would it be compressed or not?

Can you explain a bit more what you mean by "volume progress"?

**Comment by VelizarVESSELINOV** [comment]
@JohnRusk Volume progress expressed in the sum of file size `[ 2.3 GiB/ 2.3 GiB]`, not only in the number of files `[118/118 files]`. As you probably know the file size is still not standardized :) Volume bytes to be transferred is more interesting for users to do mental ETA estimation or capacity/cost estimation than file count. Egress/ingress cost/threshold are not using file count.

For the compression, expect the same/similar behavior than gsutil rsync -z or Linux rsync -j. Suggestion to study better what alternative solutions are doing around when you are building your Azure rsync tool.

Happy to hear that parallel execution is there.

**Comment by JohnRusk** [comment]
Thanks.  FYI, the percent complete figure is already progress as sum of file size.  It's bytes transferred divided by total (known) bytes. (I say known, because when scanning is still underway, the known byte count changes).  I take your point that it may not be obvious that it's done that way.

The difficulty with doing compression is this:  as far as I know, there's nothing on the Azure side to decompress the data. So if AzCopy sends the data in compressed form then it will say in compressed form. (So anything that reads it in Azure, e.g. if you read the blob from an Azure VM, will get the compressed form not the original form).  I believe that's different from rsync compression, where the file is compressed in transit, but is saved at the destination in the original uncompressed form.  For that reason, we do not currently have plans to add compression to AzCopy.

**Comment by gapra-msft** [comment]
Hi @VelizarVESSELINOV are you still looking for either of these features? We have added quite a few features since this issue was opened including the --compress cli option and out % complete does indicate % in bytes.

**Comment by gapra-msft** [comment]
Closing this issue due to inactivity. Please feel free to open a new issue if you are experiencing issues with the latest AzCopy
